src/lib/globalState/activeProject.ts

```diff
@@ -0,0 +1,207 @@
+/**
+ * @file
+ * The active project is just an instance of the class `Project` that contains everything we need to display and interact with the automata being created.
+ *
+ * There is a store `project` that allows for direct access to this `Project` class, but if everything used it,
+ * then any minor update to the project would require a full update to the entire UI.
+ * Therefore we have a scoped store for each project member, which will not update when other scoped stores update. We also have some custom stores.
+ *
+ * The store scope looks like:
+ *
+ * project
+ *  - name
+ *  - components
+ *     - activeView
+ *  - systems
+ *     - activeView
+ *  - queries
+ *  - systemDeclarations
+ *  - globalDeclarations
+ *
+ * An update to a store will trigger updates in all parent stores and child stores, but will not trigger updates in any sibling stores.
+ *
+ */
+
+import type { Writable, Subscriber, Unsubscriber, Updater } from "svelte/store";
+import type { Project } from "$lib/classes/project/Project";
+
+let projectStore: Project | undefined;
+
+const projectSubscribers = new Set<Subscriber<Project | undefined>>();
+const memberSubscribers: {
+	[key in keyof Project]: Set<Subscriber<Project[key] | undefined>>;
+} = {
+	name: new Set(),
+	components: new Set(),
+	systems: new Set(),
```

Comment:
Skal man kunne "subscribe" til individuelle "components" og "sets"

Reply:
No, this is only for internal use to keep track of the subscribers that subscribe to the members of the `Project`

Reply:
You cannot subscribe to an individual component, except in the case where you are subscribing to the active view. I don't see any other case where a specific subscribe would be necessary.

Reply:
I assume this is resolved, otherwise let me know